src/grasp_generator.cpp

```diff
@@ -1032,61 +1040,126 @@ bool GraspGenerator::generateSuctionGrasps(const Eigen::Affine3d& cuboid_top_pos
     {
-      for (double x = x_min; x <= x_max; x += x_inc)
-      {
-        for (double yaw = yaw_min; yaw < yaw_max; yaw += yaw_inc)
-        {
-          Eigen::Affine3d grasp_pose = Eigen::Affine3d::Identity();
-          grasp_pose *= Eigen::AngleAxisd(yaw, Eigen::Vector3d::UnitZ());
-          grasp_pose.translation() = Eigen::Vector3d(x, y, z);
+      Eigen::Affine3d grasp_pose = Eigen::Affine3d::Identity();
+      grasp_pose *= Eigen::AngleAxisd(yaw, Eigen::Vector3d::UnitZ());
 
-          grasp_pose = cuboid_center_top_grasp.rotation() * grasp_pose;
-          grasp_pose.translation() += cuboid_center_top_grasp.translation();
+      grasp_pose = grasp_poses[i].rotation() * grasp_pose;
+      grasp_pose.translation() += grasp_poses[i].translation();
 
-          if (debug_top_grasps_)
-            visual_tools_->publishAxis(grasp_pose, rviz_visual_tools::MEDIUM, "after_transform");
+      grasp_poses.push_back(grasp_pose);
+    }
+  }
 
-          addGrasp(grasp_pose, grasp_data, grasp_candidates, cuboid_top_pose, 0);
-        }
-      }
+  // Add Depth grasps (Z-axis)
+  num_grasps = grasp_poses.size();
+  for (std::size_t i = 0; i < num_grasps; ++i)
+  {
+    for (double z = z_min; z <= z_max; z += z_increment)
+    {
+      Eigen::Affine3d grasp_pose = Eigen::Affine3d::Identity();
+      grasp_pose.translation() = Eigen::Vector3d(0, 0, z);
+
+      grasp_pose = grasp_poses[i].rotation() * grasp_pose;
+      grasp_pose.translation() += grasp_poses[i].translation();
+
+      grasp_poses.push_back(grasp_pose);
+    }
+  }
+
+  // Add Y translation grasps
+  num_grasps = grasp_poses.size();
+  for (std::size_t i = 0; i < num_grasps; ++i)
+  {
+    for (double y = xy_min; y <= xy_max; y += xy_increment)
+    {
+      Eigen::Affine3d grasp_pose = Eigen::Affine3d::Identity();
+      grasp_pose.translation() = Eigen::Vector3d(0, y, 0);
+
+      grasp_pose = grasp_poses[i].rotation() * grasp_pose;
+      grasp_pose.translation() += grasp_poses[i].translation();
+
+      grasp_poses.push_back(grasp_pose);
+    }
+  }
+
+  // Add X translation grasps
+  num_grasps = grasp_poses.size();
+  for (std::size_t i = 0; i < num_grasps; ++i)
+  {
+    for (double x = xy_min; x <= xy_max; x += xy_increment)
+    {
+      Eigen::Affine3d grasp_pose = Eigen::Affine3d::Identity();
+      grasp_pose.translation() = Eigen::Vector3d(x, 0, 0);
+
+      grasp_pose = grasp_poses[i].rotation() * grasp_pose;
+      grasp_pose.translation() += grasp_poses[i].translation();
+
+      grasp_poses.push_back(grasp_pose);
+    }
+  }
+
+  num_grasps = grasp_poses.size();
+  for (std::size_t i = 0; i < num_grasps; ++i)
+  {
+    addGrasp(grasp_poses[i], grasp_data, grasp_candidates, cuboid_top_pose, 0);
+    if (debug_top_grasps_)
+    {
+      visual_tools_->publishAxis(grasp_poses[i], rviz_visual_tools::MEDIUM, "pose");
+      Eigen::Affine3d ideal_copy(ideal_grasp_pose_);
+      ideal_copy.translation() = Eigen::Vector3d(0.7, 0.3, 1.25);
```

Comment:
Where do these values come from? Looks like they should be parameters

Reply:
Just some random translation to make it easy to see. I have made it more clear and less magical